### PR TITLE
Feat/68 track options 3 dots

### LINKED ISSUE #68: track options (3 dots)
- add context menu
  - go to album
  - not implemented items yet
    - Add to Queue
    - Add to Playlist
    - Share track
    - Go to contributors
    - More information
  - We could add a link to a github issue for the things that aren't implemented yet

### PR DESCRIPTION
**Pull request by bjornarhagen**
Closes #68

### CODE REVIEW

File: components/track/TrackList.vue [hunk 1]
```diff
@@ -30,8 +61,15 @@ const { setCurrentSong } = inject(MediaPlaylistInjectionKey)!;
...
         show-thumbnail
-        @click="setCurrentSong(track.media?.[0]?.files?.[0]?.url || '')"
-      />
+        @play-track="setCurrentSong(track.media?.[0]?.files?.[0]?.url || '')"
+        @open-options="toggleDropdownForTrack(track.id || 0)"
+      >
+        <DropdownMenu
+          v-if="showDropDownForTrackId === track.id"
+          :items="dropdownMenuItemsForTrack(track)"
+          @close="toggleDropdownForTrack(track.id || 0)"
```
Comment (SimonSimCity) [inline components/track/TrackList.vue:70]: Taken the case you have one track twice in your playlist. Would this not open/close it for two items on a list?

File: plugins/3.mediaPlayer.ts [hunk 2]
```diff
@@ -95,5 +96,9 @@ export default defineNuxtPlugin((nuxtApp) => {
         ended.value = true;
       });
     },
+    addSongToQueue(src) {
+      // TODO
```
Comment (SimonSimCity) [inline plugins/3.mediaPlayer.ts:100]: Nice, I'll pick this up from there 👍

File: components/track/TrackList.vue [hunk 3]
```diff
@@ -12,7 +13,37 @@ withDefaults(
...
+  } else {
+    showDropDownForTrackId.value = trackId;
+  }
+};
+
+const dropdownMenuItemsForTrack = (track: TrackModel): DropdownMenuItem[] => {
+  const items = [];
+
+  if (track?.meta?.parent?.id) {
+    items.push({ text: "Go to album", link: `/album/${track.meta.parent.id}` });
```
Comment (SimonSimCity) [inline components/track/TrackList.vue:31]: The link might have to change after my PR #88 is merged. I can update this when either one is merged.

**Review by SimonSimCity — CHANGES_REQUESTED** [review]
One possible bug, please have a look and confirm whether it would be a problem.

File: components/track/TrackList.vue (hunk 1 above)
Comment (bjornarhagen) [reply components/track/TrackList.vue:70]: @SimonSimCity oh yeah, thats true. I'll fix it :)

Comment (bjornarhagen) [reply components/track/TrackList.vue]: @SimonSimCity Should be good now 👍

**Pull request merged**